lib/inferno/dsl/fhirpath_evaluation.rb

```diff
@@ -0,0 +1,135 @@
+module Inferno
+  module DSL
+    # This module contains the methods needed to configure and perform FHIRPath evaluations
+    # on FHIR resources. The actual evaluation is typically performed by an external
+    # FHIRPath evaluation service.
+    #
+    # Tests can leverage the evaluation functionality by  calling `evaluate_fhirpath` to retrieve
+    # results of FHIRPath expressions.
+    #
+    # @example
+    #
+    #   fhirpath_evaluator url: 'http://example.com/fhirpath'
+    #
+    #   fhirpath_evaluator name: :custom, url: 'http://example.com/custom-fhirpath'
+    #
+    #   results = evaluate_fhirpath(patient_resource, 'Patient.name.given', :custom)
+    #
+    # results will be an array representing the result of evaluating the given
+    # expression against the given root element.  Each "result" in the returned
+    # array will be in the form `{ "type": "[FHIR datatype name]", "element": [JSON representation of element] }`.
```

Comment:
Would it be possible/desireable to instantiate the element from FHIR::Models rather than than using the JSON representation of the element?

Reply:
But why doesn't representing that value as a FHIR model make sense, while representing it as JSON does make sense?

Reply:
The value of element is not always a json. I should update the description of the `element` key because it is confusing.  The description should be something like `primitive value or JSON representation of FHIR element` or simply the `result value of the FHIRPath expression`. 

These are the examples of results depending on the expression we are evaluating:
```
path = Patient.gender='female'
result =  [ {"type": "boolean", "element": true }]

path = Patient.name.given
result = [
    {
        "type": "string",
        "element": "patient"
    },
    {
        "type": "string",
        "element": "sample"
    }
]

path = Patient.name
result = [
    {
        "type": "HumanName",
        "element": {
            "use": "official",
            "family": "Chalmers",
            "given": [
                "patient",
                "sample"
            ]
        }
    }
]
```
I am not sure if you want to add condition handling to transform to FHIR model when the value is a fhir element.

Reply:
Right, it isn't always JSON. You always have to check the type. 

We use `fhir_models` widely throughout Inferno, so given these two options:
- Return a primitive or a Hash for complex types
- Return a primitive or a FHIR model instance for complex types

I'm wondering if there are technical reasons not to return a FHIR models instance here.

We generally expect (and encourage) test authors to use FHIR models to represent resources (e.g. in the first parameter of `evaluate_fhirpath`), so why not use them here?

Reply:
updated